R/rd-data.R

```diff
@@ -39,6 +39,60 @@ as_data.tag_item <- function(x, ...) {
 
 parse_section <- function(x, title, ...) {
   text <- flatten_para(x, ...)
+  html <- xml2::read_html(text)
+  r_blocks <- xml2::xml_find_all(html, "//div[contains(@class, 'sourceCode r')]/pre/code")
+  # now add blocks with no language information
+  nolang_r_blocks <- xml2::xml_find_all(
+    xml2::xml_find_all(html, "//pre/code/parent::*[not(ancestor::div)]"), # not in a div
+    "code"
+  )
+
+  highlight_r_block <- function(block) {
+    out <- downlit::highlight(
+      xml2::xml_text(block),
+      classes = downlit::classes_pandoc()
+    )
+    if (!is.na(out)) {
```

Comment:
This probably only happens when the code is not valid R code (e.g. a YAML block with no language information) but I suppose it could be some sort of pseudo-code, so hard to warn?

Reply:
Or there could be a warning, and the workaround for non-valid pseudo-code would be to use

````
```pseudor
bla
```
````